pkg/virt-controller/watch/vm.go

```diff
@@ -271,7 +271,110 @@ func (c *VMController) execute(key string) error {
 		return err
 	}
 
-	return createErr
+	if createErr != nil {
+		return createErr
+	}
+
+	err = handleVMRenaming(vm, c, logger)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func handleVMRenaming(vm *virtv1.VirtualMachine, c *VMController, logger *log.FilteredLogger) error {
+	// Check if renaming annotation exists
+	if vm.ObjectMeta.Annotations == nil {
+		return nil
+	}
+
+	renameTo, hasRenameToAnnotation := vm.ObjectMeta.Annotations[virtv1.RenameToAnnotation]
+
+	// Handle source VM
+	if hasRenameToAnnotation {
+		// If a VM with the new name exists, remove annotation
+		_, err := c.clientset.VirtualMachine(vm.Namespace).Get(renameTo, &v1.GetOptions{})
+
+		if err != nil {
+			if !strings.Contains(err.Error(), "not found") {
+				errMsg :=
+					fmt.Sprintf("can't determine existence of newVM: %s while processing rename of oldVM: %s",
+						renameTo, vm.Name)
+				logger.Reason(err).Error(errMsg)
+				return err
+			} else {
+				// New vm is not found, create it
+				newVM := &virtv1.VirtualMachine{
+					TypeMeta: vm.TypeMeta,
+					ObjectMeta: v1.ObjectMeta{
```

Comment:
I think that you should deep copy the given vm object and just change the name on it. Otherwise, if any new fields will be added we risk in missing them

Reply:
I agree, done